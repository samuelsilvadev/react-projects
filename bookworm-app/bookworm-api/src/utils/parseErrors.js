module.exports = function parseErrors(errors) {
	const result = {};
	if (errors) {
		Object
			.entries(errors)
			.forEach((value) => {
				result[value[0]] = value[1].message;
			});
	}
	return result;
};
