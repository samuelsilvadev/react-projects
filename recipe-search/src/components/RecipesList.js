import React from 'react';

import Card from './Card';

import './RecipesList.css';

const _renderRecipe = (recipe) => (
	<li key={recipe.recipe_id}>
		<Card
			imageSrc={recipe.image_url}
			title={recipe.title}
			subtitle={<>Publisher: <span>{recipe.publisher}</span></>}
			action={{
				text: 'View recipe',
				onClick: () => { }
			}}
		/>
	</li>
);

const RecipesList = ({ recipes = [] }) => (
	<ul className="recipe-list">
		{recipes.map(_renderRecipe)}
	</ul >
);

export default RecipesList;
