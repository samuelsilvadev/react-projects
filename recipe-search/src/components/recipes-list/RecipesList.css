.recipe-list {
	display: grid;
	grid-template-columns: repeat(3, 1fr);
	grid-row-gap: 50px;
	grid-column-gap: 10px;
	list-style: none;
	padding: 0 10px;
}
