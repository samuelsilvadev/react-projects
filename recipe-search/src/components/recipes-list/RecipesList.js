import React from 'react';
import PropTypes from 'prop-types';

import Card from '../card';

import './RecipesList.css';

const _renderRecipe = (recipe) => (
	<li key={recipe.recipe_id} data-test="recipe-list__item">
		<Card
			imageSrc={recipe.image_url}
			title={recipe.title}
			subtitle={<>Publisher: <span>{recipe.publisher}</span></>}
			link={{
				text: 'View recipe',
				to: {
					pathname: `/recipe/${recipe.recipe_id}`,
					state: {
						recipe
					}
				}
			}}
			data-test="recipe-list__card"
		/>
	</li>
);

const RecipesList = ({ recipes = [] }) => (
	<ul className="recipe-list" data-test="recipe-list">
		{recipes.map(_renderRecipe)}
	</ul >
);

RecipesList.propTypes = {
	recipes: PropTypes.arrayOf(PropTypes.shape({
		recipe_id: PropTypes.number.isRequired,
		image_url: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		publisher: PropTypes.string.isRequired
	}))
};

export default RecipesList;
