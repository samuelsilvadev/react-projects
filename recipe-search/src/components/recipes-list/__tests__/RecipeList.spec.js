import React from 'react';
import { shallow } from 'enzyme';

import RecipesList from '../RecipesList';

import { toTestId } from '@tests/utils';

const recipeMock = {
	recipe_id: 123456,
	image_url: 'hey.png',
	title: 'Pasta',
	publisher: 'Yo miesmo'
};

const recipesMock = [recipeMock];

describe('<RecipeList />', () => {
	it('should render correctly when has recipes', () => {
		const wrapper = shallow(<RecipesList recipes={recipesMock} />);

		['recipe-list', 'recipe-list__item', 'recipe-list__card']
			.forEach(dataTest =>
				expect(wrapper.find(toTestId(dataTest)).exists()).toBe(true)
			);

		expect(wrapper).toMatchSnapshot();
	});

	it('should render correctly when has no recipes', () => {
		const wrapper = shallow(<RecipesList />);

		expect(wrapper.find(toTestId('recipe-list')).exists()).toBe(true);

		['recipe-list__item', 'recipe-list__card']
			.forEach(dataTest =>
				expect(wrapper.find(toTestId(dataTest)).exists()).toBe(false)
			);

		expect(wrapper).toMatchSnapshot();
	});
});
