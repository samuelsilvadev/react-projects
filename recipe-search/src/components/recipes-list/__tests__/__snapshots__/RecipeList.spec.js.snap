// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeList /> should render correctly when has no recipes 1`] = `
<ul
  className="recipe-list"
  data-test="recipe-list"
/>
`;

exports[`<RecipeList /> should render correctly when has recipes 1`] = `
<ul
  className="recipe-list"
  data-test="recipe-list"
>
  <li
    data-test="recipe-list__item"
    key="123456"
  >
    <Card
      data-test="recipe-list__card"
      imageSrc="hey.png"
      link={
        Object {
          "text": "View recipe",
          "to": Object {
            "pathname": "/recipe/123456",
            "state": Object {
              "recipe": Object {
                "image_url": "hey.png",
                "publisher": "Yo miesmo",
                "recipe_id": "123456",
                "title": "Pasta",
              },
            },
          },
        }
      }
      subtitle={
        <React.Fragment>
          Publisher: 
          <span>
            Yo miesmo
          </span>
        </React.Fragment>
      }
      title="Pasta"
    />
  </li>
</ul>
`;
